src/delay.rs

```diff
@@ -10,12 +10,13 @@ use time::Hertz;
 
 /// System timer (SysTick) as a delay provider
 pub struct Delay {
-    sysclk: Hertz,
+    sysclk: Hertz
     syst: SYST,
 }
 
 impl Delay {
     /// Configures the system timer (SysTick) as a delay provider
+    // AJM - Consistency: stm32f hal takes ownship of clocks here, where you only take/clone the sysclk `Hertz` object
```

Comment:
Yeah, japaric marks his clocks object as Copy/Clone. I don't so I pass it by reference. Why would you want your own copy of the clock info - it can't/shouldn't change?